contrib/storage-jdbc/src/main/java/org/apache/drill/exec/store/jdbc/JdbcStoragePlugin.java

```diff
@@ -108,19 +109,22 @@ public Optional<DataSource> getDataSource(UserCredentials userCredentials) {
     ));
   }
 
-  public SqlDialect getDialect(DataSource dataSource) {
-    return JdbcSchema.createDialect(
-      SqlDialectFactoryImpl.INSTANCE,
-      dataSource
-    );
+  public synchronized SqlDialect getDialect(UserCredentials userCredentials) {
+    if (sqlDialect == null) {
```

Comment:
Please use double-check to avoid extra overhead caused by synchronization for most of the calls when the  `sqlDialect` is already created.